### PR TITLE
Add documentation on connecting remote applications

### PR DESCRIPTION
**Pull request by RukshiW**
Fixes https://github.com/wso2-enterprise/choreo/issues/6735

### CODE REVIEW

File: en/docs/remote-apps/connect-remote-app.md [hunk 1]
```diff
@@ -0,0 +1,123 @@
...
+    
+    ```
+    ballerina: initializing connection with observability backend periscope.choreo.dev:443
+    ballerina: visit http://console.choreo.dev/observe/app/91e9750f-e2e1-11eb-90ce-de6d84510939/dd9780a8-e2e2-11eb-90ce-de6d84510939 to access observability data
+    ballerina: started publishing metrics to Choreo
+    [ballerina/http] started HTTP/WS listener 0.0.0.0:9090
+    ballerina: started publishing traces to Choreo
+    ballerina: sending all remaining traces to Choreo
+    [ballerina/http] stopped HTTP/WS listener 0.0.0.0:9090
+    ballerina: sending metrics to Choreo
```
Comment (sachiniSam) [inline en/docs/remote-apps/connect-remote-app.md:90]: we may need to remove these three lines, as they are executed once we stop the Ballerina service.
In Step 4, we mention about invoking the service, therefore its better to have the output at the running state.

Comment (RukshiW) [reply en/docs/remote-apps/connect-remote-app.md]: Done

File: en/docs/remote-apps/connect-remote-app.md [hunk 2]
```diff
@@ -0,0 +1,120 @@
+# Connect Remote Applications
+
+If you need to observe an external application that you have written in Ballerina programming language, you can connect it to Choreo.
+
+## Prerequisites
+
+Create an external application in Ballerina programming language and run it. For instructions, see [Ballerina Documentation - Writing Your First Ballerina Program](https://ballerina.io/learn/user-guide/getting-started/writing-your-first-ballerina-program/).
```
Comment (sachiniSam) [inline en/docs/remote-apps/connect-remote-app.md:7]: Under per-requisites, i think we dont need to mention to `run` the Ballerina program. With the step 3 we mention how to build and run.
Having a Ballerina project would be enough for the user to connect to Choreo

File: en/docs/remote-apps/connect-remote-app.md [hunk 3]
```diff
@@ -0,0 +1,120 @@
...
+
+In this example, you can invoke the `HelloService` application via the following cURL command.
+
+```text
+curl http://localhost:9090/hello/sayHello
+```
+
+After sending a few requests, you can proceed to the next step to view the observability statistics generated for your Ballerina application.
+
+## Step 5: View observability statistics
```
Comment (sachiniSam) [inline en/docs/remote-apps/connect-remote-app.md:104]: To view the Observability stats, User could also click on the link provided in the terminal as given in the sample in Step 3. 
```
ballerina: visit http://console.choreo.dev/observe/app/91e9750f-e2e1-11eb-90ce-de6d84510939/dd9780a8-e2e2-11eb-90ce-de6d84510939 to access observability data
```
Maybe we could add that approach under Step 5 as well.

File: en/docs/remote-apps/connect-remote-app.md (hunk 2 above)
Comment (RukshiW) [reply en/docs/remote-apps/connect-remote-app.md]: Done

File: en/docs/remote-apps/connect-remote-app.md (hunk 3 above)
Comment (RukshiW) [reply en/docs/remote-apps/connect-remote-app.md:104]: Done

File: en/mkdocs.yml [hunk 4]
```diff
@@ -61,6 +61,8 @@ nav:
     - Integrations:
           - Integration Concepts: integrations/integration-concepts.md
           - Build Your First Integration: integrations/build-your-first-integration.md
+    - Remote Apps:
+          - Connect Remote Applications: remote-apps/connect-remote-app.md
```
Comment (anupama-pathirage) [inline en/mkdocs.yml:65]: Always use the same name as the topic to the file name. `connect-remote-application.md`

File: en/docs/remote-apps/connect-remote-app.md [hunk 5]
```diff
@@ -0,0 +1,118 @@
...
+
+    ```text
+    bal build
+    ```
+   
+    This creates a `JAR` in the `<BALLERINA_PROJECT>/target/bin` sub directory.
+   
+2. To run the Ballerina application, issue the following command from the `<BALLERINA_PROJECT>`.
+
+    ```
```
Comment (anupama-pathirage) [inline en/docs/remote-apps/connect-remote-app.md:74]: We better use `bash` here

File: en/docs/remote-apps/connect-remote-app.md [hunk 6]
```diff
@@ -0,0 +1,118 @@
+# Connect Remote Applications
+
+If you need to observe an external application that you have written in Ballerina programming language, you can connect it to Choreo.
```
Comment (anupama-pathirage) [inline en/docs/remote-apps/connect-remote-app.md:3]: Link ballerina.io

File: en/docs/remote-apps/connect-remote-app.md [hunk 7]
```diff
@@ -0,0 +1,118 @@
...
+
+After sending a few requests, you can proceed to the next step to view the observability statistics generated for your Ballerina application.
+
+## Step 5: View observability statistics
+
+To view the observability statistics generated for your remote Ballerina application, click on the link provided for observability data in the startup logs of your remote application. Ah shown in the startup logs generated in [Step 3](step-3-build-the-ballerina-project), the link in this example is `http://console.choreo.dev/observe/app/91e9750f-e2e1-11eb-90ce-de6d84510939/dd9780a8-e2e2-11eb-90ce-de6d84510939`.
+
+Observability data is displayed as shown in the example below.
+
+![View observability data](../assets/img/remote-apps/remote-service-observability.png)
```
Comment (anupama-pathirage) [inline en/docs/remote-apps/connect-remote-app.md:110]: Image styles are missing

File: en/docs/remote-apps/connect-remote-app.md [hunk 8]
```diff
@@ -0,0 +1,118 @@
+# Connect Remote Applications
+
+If you need to observe an external application that you have written in Ballerina programming language, you can connect it to Choreo.
+
+## Prerequisites
+
+Create an external application in Ballerina programming language. For instructions, see [Ballerina Documentation - Writing Your First Ballerina Program](https://ballerina.io/learn/user-guide/getting-started/writing-your-first-ballerina-program/).
```
Comment (anupama-pathirage) [inline en/docs/remote-apps/connect-remote-app.md:7]: Why do we say external here. IMO, `Create an application...` is enough as external can lead to different meanings

File: en/docs/remote-apps/connect-remote-app.md [hunk 9]
```diff
@@ -0,0 +1,118 @@
...
+
+## Step 5: View observability statistics
+
+To view the observability statistics generated for your remote Ballerina application, click on the link provided for observability data in the startup logs of your remote application. Ah shown in the startup logs generated in [Step 3](step-3-build-the-ballerina-project), the link in this example is `http://console.choreo.dev/observe/app/91e9750f-e2e1-11eb-90ce-de6d84510939/dd9780a8-e2e2-11eb-90ce-de6d84510939`.
+
+Observability data is displayed as shown in the example below.
+
+![View observability data](../assets/img/remote-apps/remote-service-observability.png)
+
+!!! tip
```
Comment (anupama-pathirage) [inline en/docs/remote-apps/connect-remote-app.md:112]:
```suggestion
!!! Tip
```

File: en/mkdocs.yml (hunk 4 above)
Comment (RukshiW) [reply en/mkdocs.yml]: Done

File: en/docs/remote-apps/connect-remote-app.md [hunk 10]
```diff
@@ -0,0 +1,118 @@
...
+   
+Now you have completed setting up your Ballerina project. It will connect to Choreo once you build and run it.
+
+## Step 3: Build the Ballerina project
+
+In this step, you will build and run your Ballerina project. 
+
+1. To build the Ballerina project, open your terminal, navigate to `<BALLERINA_PROJECT>`, and issue the following command.
+
+    ```text
```
[the remark this answers is not in the document]
Comment (RukshiW) [reply en/docs/remote-apps/connect-remote-app.md]: Done

File: en/docs/remote-apps/connect-remote-app.md (hunk 5 above)
Comment (RukshiW) [reply en/docs/remote-apps/connect-remote-app.md]: Done

File: en/docs/remote-apps/connect-remote-app.md [hunk 11]
```diff
@@ -0,0 +1,118 @@
...
+
+    ```
+    bal run target/bin/<JAR_NAME>.jar
+    ```
+   
+    e.g., `bal run target/bin/HelloService.jar`
+    
+    Logs similar to the following appear in the terminal.
+    
+    ```
```
[the remark this answers is not in the document]
Comment (RukshiW) [reply en/docs/remote-apps/connect-remote-app.md]: Done

File: en/docs/remote-apps/connect-remote-app.md (hunk 6 above)
Comment (RukshiW) [reply en/docs/remote-apps/connect-remote-app.md]: Done

File: en/docs/remote-apps/connect-remote-app.md (hunk 7 above)
Comment (RukshiW) [reply en/docs/remote-apps/connect-remote-app.md]: Added image style.

File: en/docs/remote-apps/connect-remote-app.md (hunk 8 above)
Comment (RukshiW) [reply en/docs/remote-apps/connect-remote-app.md]: Removed the word "external"

File: en/docs/remote-apps/connect-remote-app.md (hunk 9 above)
Comment (RukshiW) [reply en/docs/remote-apps/connect-remote-app.md]: Done.

File: en/docs/remote-apps/connect-remote-applications.md [hunk 12]
```diff
@@ -0,0 +1,118 @@
+# Connect Remote Applications
+
+If you need to observe an external application that you have written in [Ballerina programming language](https://ballerina.io/), you can connect it to Choreo.
```
Comment (Nashaath) [inline en/docs/remote-apps/connect-remote-applications.md:3]:
```suggestion
If you need to observe an external application that you have written in the [Ballerina programming language](https://ballerina.io/), you can connect it to Choreo.
```

File: en/docs/remote-apps/connect-remote-applications.md [hunk 13]
```diff
@@ -0,0 +1,118 @@
...
+
+## Prerequisites
+
+Create an application in Ballerina programming language. For instructions, see [Ballerina Documentation - Writing Your First Ballerina Program](https://ballerina.io/learn/user-guide/getting-started/writing-your-first-ballerina-program/).
```
Comment (Nashaath) [inline en/docs/remote-apps/connect-remote-applications.md:7]:
```suggestion
Create an application in the Ballerina programming language. For instructions, see [Ballerina Documentation - Writing Your First Ballerina Program](https://ballerina.io/learn/user-guide/getting-started/writing-your-first-ballerina-program/).
```

File: en/docs/remote-apps/connect-remote-applications.md [hunk 14]
```diff
@@ -0,0 +1,118 @@
...
+
+!!! Tip
+    In this guide, we are using the same Ballerina application that is created in the guide linked above.
+
+## Step 1: Generate and copy the Ballerina configurations
```
Comment (Nashaath) [inline en/docs/remote-apps/connect-remote-applications.md:12]: Should we say "Generate and copy the Ballerina configuration"?

File: en/docs/remote-apps/connect-remote-applications.md [hunk 15]
```diff
@@ -0,0 +1,118 @@
...
+
+6. Copy the Ballerina configurations.
+
+Now you have the application secret that you need to set up the Ballerina project of your remote application.
+
+## Step 2: Set up the Ballerina project
+
+In this step, you will set up your Ballerina project with the configurations that connect the Ballerina application to Choreo. Here, the Ballerina project is the directory that contains the definition of your Ballerina application in a `.bal` file and all the configuration files required to run it.
+
+To set up the Ballerina Project:
```
Comment (Nashaath) [inline en/docs/remote-apps/connect-remote-applications.md:36]: We have not capitalized the P with regard to the term Project in the heading and para above. So I guess we can avoid the capitalization here too.

File: en/docs/remote-apps/connect-remote-applications.md [hunk 16]
```diff
@@ -0,0 +1,118 @@
...
+
+1. Create a directory in your machine (e.g., named `HelloWorld`) and move your file with the remote ballerina application (in this example the `hello_service.bal` file) to it.
```
Comment (Nashaath) [inline en/docs/remote-apps/connect-remote-applications.md:38]: Need to capitalize B in "Ballerina".

File: en/docs/remote-apps/connect-remote-applications.md [hunk 17]
```diff
@@ -0,0 +1,118 @@
...
+
+In this step, you will build and run your Ballerina project. 
+
+1. To build the Ballerina project, open your terminal, navigate to `<BALLERINA_PROJECT>`, and issue the following command.
+
+    ```bash
+    bal build
+    ```
+   
+    This creates a `JAR` in the `<BALLERINA_PROJECT>/target/bin` sub directory.
```
Comment (Nashaath) [inline en/docs/remote-apps/connect-remote-applications.md:70]: subdirectory should be one word.

File: en/docs/remote-apps/connect-remote-applications.md [hunk 18]
```diff
@@ -0,0 +1,118 @@
...
+
+```bash
+curl http://localhost:9090/hello/sayHello
+```
+
+After sending a few requests, you can proceed to the next step to view the observability statistics generated for your Ballerina application.
+
+## Step 5: View observability statistics
+
+To view the observability statistics generated for your remote Ballerina application, click on the link provided for observability data in the startup logs of your remote application. Ah shown in the startup logs generated in [Step 3](step-3-build-the-ballerina-project), the link in this example is `http://console.choreo.dev/observe/app/91e9750f-e2e1-11eb-90ce-de6d84510939/dd9780a8-e2e2-11eb-90ce-de6d84510939`.
```
Comment (Nashaath) [inline en/docs/remote-apps/connect-remote-applications.md:106]: Better to remove the term "on" in "click on the link provided for observability data......."

"Ah" seems to be a typo. Should be changed to "As".

File: en/docs/remote-apps/connect-remote-applications.md [hunk 19]
```diff
@@ -0,0 +1,118 @@
...
+2. To link the Ballerina application to Choreo via the application secret, open a new file in a text editor and add the Ballerina configurations you copied in Step 1. Then save it as `Config.toml` in the `<BALLERINA_PROJECT>`.
+
+3. To enable the Ballerina application to publish observability statistics, save a file named `Ballerina.toml` with the following content in the `<BALLERINA_PROJECT>`.
+
+    ```ballerina
+    [build-options]
+    observabilityIncluded = true
+    ```
+   
+4. For the Ballerina application to import the `ballerinax/choreo` module, open the `.bal` file (in this example, `hello_service.bal`) and at the following line at the top.
```
Comment (Nashaath) [inline en/docs/remote-apps/connect-remote-applications.md:52]: Should be "add the following line. . . ." right?

**Review by Nashaath — CHANGES_REQUESTED** [review]
Added a few minor suggestions and comments.

File: en/docs/remote-apps/connect-remote-applications.md (hunk 14 above)
Comment (RukshiW) [reply en/docs/remote-apps/connect-remote-applications.md]: Changed.

File: en/docs/remote-apps/connect-remote-applications.md (hunk 15 above)
Comment (RukshiW) [reply en/docs/remote-apps/connect-remote-applications.md]: Corrected

File: en/docs/remote-apps/connect-remote-applications.md (hunk 16 above)
Comment (RukshiW) [reply en/docs/remote-apps/connect-remote-applications.md]: Done

File: en/docs/remote-apps/connect-remote-applications.md (hunk 17 above)
Comment (RukshiW) [reply en/docs/remote-apps/connect-remote-applications.md]: Corrected

File: en/docs/remote-apps/connect-remote-applications.md (hunk 18 above)
Comment (RukshiW) [reply en/docs/remote-apps/connect-remote-applications.md]: Corrected

File: en/docs/remote-apps/connect-remote-applications.md (hunk 19 above)
Comment (RukshiW) [reply en/docs/remote-apps/connect-remote-applications.md]: Corrected

**Review by Nashaath — APPROVED** [review]

**Pull request merged**